src/variables/components/VariableDropdownReactSelect.tsx

```diff
@@ -0,0 +1,369 @@
+
+    const onChange = selection => {
+      if (selection && selection.value) {
+        this.handleSelect(selection.value)
+      }
+    }
+    // primary50 is the color that is used when an item is clicked on to change the selection
+    // have not found an alternative way (via the customStyles) to change that color)
+
+    //    const style={width:200}
+    const selectedObjectValue = {value: selectedValue, label: selectedValue}
+    return (
+      <Select
+        classNamePrefix="select"
+        onChange={onChange}
+        defaultValue={selectedObjectValue}
+        isDisabled={isDisabled}
+        isLoading={status === RemoteDataState.Loading || isDisabled}
+        isClearable={true}
+        isRtl={false}
+        isSearchable={true}
+        styles={customStyles}
+        name={name}
+        options={realVals}
+        placeholder={placeHolderText}
+        theme={theme => ({
+          ...theme,
+          colors: {
+            ...theme.colors,
+            primary50: selectColor,
+          },
+        })}
+      />
+    )
+    //
+  }
+
+  // private getWidth(placeHolderText) {
+  //   const {values} = this.props
+  //   const allVals = [placeHolderText, ...values]
+  //   const longestItemWidth = Math.floor(
+  //     allVals.reduce(function(a, b) {
+  //       return a.length > b.length ? a : b
+  //     }, '').length * 10
+  //   )
+  //
+  //   const widthLength = Math.max(192, longestItemWidth)
+  //   const widthStyle = {width: `${widthLength}px`}
+  //   return widthStyle
+  // }
+
+  private handleSelect = (selectedValue: string, closeMenuNow?: boolean) => {
+    const {
+      variableID,
+      onSelectValue,
+      onSelect,
+      selectedValue: prevSelectedValue,
+    } = this.props
+
+    if (prevSelectedValue !== selectedValue) {
+      onSelectValue(variableID, selectedValue)
+    }
+
+    if (onSelect) {
+      onSelect()
+    }
+
+    const newState = {
+      ...this.state,
+      typedValue: selectedValue,
+      actualVal: selectedValue,
+      selectHappened: true,
+      selectIndex: -1,
+    }
+
+    if (closeMenuNow) {
+      newState['menuOpen'] = MenuStatus.Closed
+    }
+
+    this.setState(newState)
+  }
+
+  // show the 'loading' or 'no values' as a string (no input field yet!)
+  // when it is loading
+  getPlaceHolderText = (defaultText: string = '') => {
+    const {status} = this.props
+
+    if (status === RemoteDataState.Loading) {
+      return 'Loading...'
+    }
+    if (this.noFilteredValuesPresent()) {
+      return 'No Values'
+    }
+
+    return defaultText
+  }
+
+  noFilteredValuesPresent = () => {
+    const {shownValues} = this.state
+    return this.internalNoValuesPresent(shownValues)
+  }
+
+  noValuesPresent = () => {
+    const {values} = this.props
+    return this.internalNoValuesPresent(values)
+  }
+
+  internalNoValuesPresent = valsToUse => {
+    const {status} = this.props
+
+    return (
+      status === RemoteDataState.Done && (!valsToUse || valsToUse.length === 0)
```

Comment:
same comment here about being a little bit safer to use

```suggestion
      status === RemoteDataState.Done && (!Array.isArray(valsToUse) || valsToUse.length === 0)
```